packages/lib/src/vis/shared/SelectionTool.tsx

```diff
@@ -57,16 +58,20 @@ function SelectionTool(props: Props) {
         return;
       }
       const point = worldToData(boundPointToFOV(evt.unprojectedPoint, camera));
-
       setEndPoint(point);
+
+      if (!modifierKeyPressed) {
+        return;
+      }
```

Comment:
Prevent the call of `onSelectionChange` if the modifier key is not pressed but still change the endpoint thanks to `setEndpoint` above.